cf_xarray/parametric.py

```diff
@@ -0,0 +1,691 @@
+    if (req & terms) != req:
+        req_diff = sorted(req - terms)
+
+        raise KeyError(
+            f"Required terms {', '.join(req_diff)} are absent in the dataset."
+        )
+
+    # if there are optional arguments check that atleast one
+    # is in the intersection, only required for
+    # atmosphere_hybrid_sigma_pressure_coordinate
+    if len(opt) > 0 and len(opt & terms) == 0:
+        raise KeyError(
+            f"Atleast one optional term {', '.join(sorted(opt))} is absent in the dataset."
+        )
+
+
+def func_from_stdname(stdname):
+    """Get function from module.
+
+    Uses `stdname` to return function from module.
+
+    Parameters
+    ----------
+    stdname : str
+        Name of the function.
+
+    Raises
+    ------
+    AttributeError
+        If a function name `stdname` is not in the module.
+    """
+    m = sys.modules[__name__]
+
+    return getattr(m, stdname)
+
+
+def derive_dimension_order(output_order, **dim_map):
+    """Derive dimension ordering from input map.
+
+    This will derive a dimensinal ordering from a map of dimension
+    identifiers and variables containing the dimensions.
+
+    This is useful when dimension names are not know.
+
+    For example if the desired output ordering was "nkji" where
+    variable "A" contains "nji" (time, lat, lon) and "B" contains
+    "k" (height) then the output would be (time, height, lat, lon).
+
+    This also works when dimensions are missing.
+
+    For example if the desired output ordering was "nkji" where
+    variable "A" contains "n" (time) and "B" contains
+    "k" (height) then the output would be (time, height).
+
+    Parameters
+    ----------
+    output_order : str
+        Dimension identifiers in desired order, e.g. "nkji".
+    **dim_map : dict
+        Dimension identifiers and variable containing them, e.g. "nji": eta, "k": s.
+
+    Returns
+    -------
+    list
+        Output dimensions in desired order.
+    """
+    dims = {}
+
+    for x, y in dim_map.items():
+        for i, z in enumerate(x):
+            try:
+                dims[z] = y.dims[i]
+            except IndexError:
+                dims[z] = None
+
+    return tuple(dims[x] for x in list(output_order) if dims[x] is not None)
+
+
+def atmosphere_ln_pressure_coordinate(p0, lev):
+    """Atmosphere natural log pressure coordinate.
+
+    Standard name: atmosphere_ln_pressure_coordinate
+
+    Parameters
+    ----------
+    p0 : xr.DataArray
+        Reference pressure.
+    lev : xr.DataArray
+        Vertical dimensionless coordinate.
+
+    Returns
+    -------
+    xr.DataArray
+        A DataArray with new pressure coordinate.
+
+    References
+    ----------
+    Please refer to the CF conventions document :
+      1. https://cfconventions.org/cf-conventions/cf-conventions.html#atmosphere-natural-log-pressure-coordinate
+    """
+    p = p0 * np.exp(-lev)
+
+    p = p.squeeze().rename("p").assign_attrs(standard_name="air_pressure")
```

Comment:
```suggestion
    p = p.squeeze().assign_attrs(standard_name="air_pressure")
```

I don't think we need to hardcode these names since we set them when assigning to the dataset in `.cf.decode_vertical_coords`